### ISSUE TITLE
question: is linting possible?

### ISSUE THREAD

**Issue by rumyantseva** [go-critic/go-critic#661]
There is an example from Bill Kenndy's workshop: https://play.golang.org/p/D88zzGYanvX

In this example, there is an error on line 34: it changes the number of likes for `shareUser`, but the number of like for `users` will not be changed. Is it possible to write a linter which might report such a situation? If it's not possible, why?

**Comment by rumyantseva** [comment]
Thanks, Iskander. I didn't find a proper title for this issue, so feel free to change the title (as well as the description) to keep this issue in a better form.

**Comment by Quasilyte** [comment]
Warnings should be given if:
1. There is a live local pointer `p` to a slice `s` element.
2. Slice `s` is potentially mutated in the scope of `p`.

To avoid false positives until global analysis (or new `go/analysis`) is used, only direct `append` can be considered as a guaranteed mutation of `s`.

This makes this code generate a warning:
```go
{
  p := &s[i]
  if someCond {
    s = append(s, x) // Might invalidate `p`
  }
}
s = append(s, y) // OK, no live pointers at this point
```

It would be good to check this implementation on a Go corpus to check whether it finds any real problems or only gives close to false positive results.
